### ISSUE TITLE
Impossible to generate OPKGC metapackage packages on Debian

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#448]
**Reported by valleegr on 10 Jan 2008 15:21 UTC**
Using the opkgc trunk, i have the following error on Debian Etch:

```
opkgc --dist debian -o /home/gvh/temp/opkgc/ -v
[Read config file /etc/opkgc.conf
[DEBUG](INFO]) /usr/local/share/opkgc/opkg.xsd loaded
[./config.xml loaded
[INFO](DEBUG]) Validating XML file against schema
[Package name: sis
[DEBUG](DEBUG]) Create dir: /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/scripts
[Copy ./scripts/si_monitor.patch to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/scripts
[DEBUG](DEBUG]) Copy ./scripts/server-post-install to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/scripts
[Copy ./scripts/api-post-clientdef to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/scripts
[DEBUG](DEBUG]) Copy ./scripts/api-post-image to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/scripts
[Copy ./opkg-sis_4.0.2.orig.tar.gz to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2
[DEBUG](DEBUG]) Copy ./configurator.html to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2
[Create dir: /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/doc
[DEBUG](DEBUG]) Copy ./doc/license.tex to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/doc
[Copy ./doc/install.tex to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/doc
[DEBUG](DEBUG]) Copy ./doc/user.tex to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/doc
[Create dir: /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2
[DEBUG](DEBUG]) Create dir: /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/scripts
[Copy ./opkg-sis-4.0.2/scripts/si_monitor.patch to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/scripts
[DEBUG](DEBUG]) Copy ./opkg-sis-4.0.2/scripts/server-post-install to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/scripts
[Copy ./opkg-sis-4.0.2/scripts/api-post-clientdef to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/scripts
[DEBUG](DEBUG]) Copy ./opkg-sis-4.0.2/scripts/api-post-image to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/scripts
[Copy ./opkg-sis-4.0.2/configurator.html to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2
[DEBUG](DEBUG]) Create dir: /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/doc
[Copy ./opkg-sis-4.0.2/doc/license.tex to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/doc
[DEBUG](DEBUG]) Copy ./opkg-sis-4.0.2/doc/install.tex to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/doc
[Copy ./opkg-sis-4.0.2/doc/user.tex to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/doc
[DEBUG](DEBUG]) Create dir: /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[Copy ./opkg-sis-4.0.2/debian/compat to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[DEBUG](DEBUG]) Copy ./opkg-sis-4.0.2/debian/opkg-server-sis.install to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[Copy ./opkg-sis-4.0.2/debian/README.debian to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[DEBUG](DEBUG]) Copy ./opkg-sis-4.0.2/debian/control to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[Copy ./opkg-sis-4.0.2/debian/rules to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[DEBUG](DEBUG]) Copy ./opkg-sis-4.0.2/debian/opkg-sis.install to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[Copy ./opkg-sis-4.0.2/debian/copyright to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[DEBUG](DEBUG]) Copy ./opkg-sis-4.0.2/debian/opkg-client-sis.install to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[Copy ./opkg-sis-4.0.2/debian/changelog to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2/debian
[DEBUG](DEBUG]) Copy ./opkg-sis-4.0.2/config.xml to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2
[Copy ./opkg-sis-4.0.2/.configurator.temp.html to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2/opkg-sis-4.0.2
[DEBUG](DEBUG]) Copy ./config.xml to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2
[Copy ./.configurator.temp.html to /tmp/tmpaw9h3c.opkgc/opkg-sis-4.0.2
[DEBUG](DEBUG]) Create tarball '/home/gvh/temp/opkgc/opkg-sis-4.0.2.tar.gz' with ' opkg-sis-4.0.2' into '/tmp/tmpaw9h3c.opkgc'
[Execute: tar zcf /home/gvh/temp/opkgc/opkg-sis-4.0.2.tar.gz  opkg-sis-4.0.2
[DEBUG](DEBUG]) Delete temp dir: /tmp/tmpaw9h3c.opkgc
[opkg tarball created: /home/gvh/temp/opkgc/opkg-sis-4.0.2.tar.gz
[DEBUG](INFO]) Untar 'opkg-sis_4.0.2.orig.tar.gz' into '/home/gvh/temp/opkgc/'
[Execute: tar zxf opkg-sis_4.0.2.orig.tar.gz
[INFO](DEBUG]) tar: opkg-sis_4.0.2.orig.tar.gz: Cannot open: No such file or directory
[tar: Error is not recoverable: exiting now
[INFO](INFO]) tar: Child returned status 2
[tar: Error exit delayed from previous errors
[DEBUG](INFO]) Templates: ['/usr/local/share/opkgc/tmpl/debian/compat', '/usr/local/share/opkgc/tmpl/debian/control.tmpl', '/usr/local/share/opkgc/tmpl/debian/changelog.tmpl', '/usr/local/share/opkgc/tmpl/debian/rules', '/usr/local/share/opkgc/tmpl/debian/README.debian.tmpl']('/usr/local/share/opkgc/tmpl/debian/copyright.tmpl',)
[Generates /home/gvh/temp/opkgc/opkg-sis-4.0.2/debian/copyright from template /usr/local/share/opkgc/tmpl/debian/copyright.tmpl
[INFO](INFO]) Copy /usr/local/share/opkgc/tmpl/debian/compat to /home/gvh/temp/opkgc/opkg-sis-4.0.2/debian
[Generates /home/gvh/temp/opkgc/opkg-sis-4.0.2/debian/control from template /usr/local/share/opkgc/tmpl/debian/control.tmpl
[DEBUG](INFO]) maintainer: 2
[uploader: 0
[DEBUG](DEBUG]) Depends: oscar-base
[Depends: systemimager-client, systemconfigurator, systemimager-common, bittorrent, systeminstaller-oscar, systeminstaller-oscar-x11, systemimager-server, systemimager-flamethrower, systemimager-bittorrent, oscar-base-server, syslinux, atftpd, systemimager-boot-amd64-standard, systemimager-boot-i386-standard, systemimager-initrd-template-amd64, systemimager-initrd-template-i386, syslinux, atftpd, systemimager-boot-i386-standard, systemimager-initrd-template-i386
[DEBUG](DEBUG]) Depends: systemconfigurator, systemimager-common, systemimager-client, oscar-base-client
[Generates /home/gvh/temp/opkgc/opkg-sis-4.0.2/debian/changelog from template /usr/local/share/opkgc/tmpl/debian/changelog.tmpl
[INFO](INFO]) Copy /usr/local/share/opkgc/tmpl/debian/rules to /home/gvh/temp/opkgc/opkg-sis-4.0.2/debian
[Generates /home/gvh/temp/opkgc/opkg-sis-4.0.2/debian/README.debian from template /usr/local/share/opkgc/tmpl/debian/README.debian.tmpl
[DEBUG](INFO]) Execute: dpkg-buildpackage -rfakeroot -sa
[parsechangelog/debian: error: badly formatted trailer line, at file debian/changelog line 6
[INFO](INFO]) dpkg-buildpackage: cannot determine name of current package
[Packages generation failed: return 9
```

The changelog file looks like:
```
$ cat /home/gvh/temp/opkgc/opkg-sis-4.0.2/debian/changelog
opkg-sis (4.0.2-1) unstable; urgency=low

  [Geoffroy Vallee](ERROR])
  - Updated systemimager to 4.0.2-1.

 --

opkg-sis (4.0.0-1) unstable; urgency=low

  [Focht](Erich)
  - Updated systemimager to 4.0.0-1.

 --

opkg-sis (3.9.6-1) unstable; urgency=low

  [Focht](Erich)
  - Updated systeminstaller to 2.3.3-1.
  - Updated systemimager to 3.9.6-1.
  - Updated systemconfigurator to 2.2.10-1

 --

opkg-sis (3.8.0-4) unstable; urgency=low

  [Focht](Erich)
  - Fixing dependencies of client and server packages.

 --

opkg-sis (3.8.0-3) unstable; urgency=low

  [Vallee](Geoffroy)
  - Fixes the dependence with atftp on Debian for the server package

 --

opkg-sis (3.8.0-2) unstable; urgency=low

  [Vallee](Geoffroy)
  - Fixes the dependence with systemimager-client

 --

opkg-sis (3.8.0-1) unstable; urgency=low

  [Vallee](Geoffroy)
  - Convert to the new OPKG API (for OPKGC)

 --
```

**Comment by dikim33** [comment]
**Comment by dikim on 10 Jan 2008 17:38 UTC**
Hi Geoffroy,

Can you try to apply the following patch and then see if it works for you?
I can not really test this because I don't have any debian yet. ;-)

```
Index: Compiler.py
===================================================================
--- Compiler.py (revision 6837)
+++ Compiler.py (working copy)
@@ -175,7 +175,7 @@
         sourcedir = os.path.join(self.dest_dir,
                                  "opkg-%s-%s" % (self.opkgName, self.opkgDesc.getVersion('upstream')))
         # Rename tar to follow Debian non-native package rule
-        debtarfile = "opkg-%s_%s.orig.tar.gz" % (self.opkgName, self.opkgDesc.getVersion('upstream'))
+        debtarfile = "opkg-%s_%s.tar.gz" % (self.opkgName, self.opkgDesc.getVersion('upstream'))
         os.rename(tarfile, debtarfile)
         
         # Uncompress tar
```

Regards,

- DongInn

**Comment by dikim33** [comment]
**Comment by jparpail on 7 Feb 2008 13:10 UTC**
(In [[OPKGC](6927]))
	Rename correctly tar file for Debian package creation  (Closes #449)
